src/main/java/com/quantum_pixel/arg/hotel/service/MailSenderService.java

```diff
@@ -0,0 +1,59 @@
+package com.quantum_pixel.arg.hotel.service;
+
+import com.quantum_pixel.arg.hotel.exception.PastDateException;
+import com.quantum_pixel.arg.hotel.model.mail.ConferenceMailStructure;
+import com.quantum_pixel.arg.hotel.model.mail.MailStructure;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.mail.SimpleMailMessage;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.mail.javamail.JavaMailSenderImpl;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDate;
+
+
+@Service
+@RequiredArgsConstructor
+public class MailSenderService {
+    private static final String email = "lukabuziu42@gmail.com";
+
+    private final JavaMailSender javaMailSender;
+
+    @Value("${spring.mail.username}")
+    private String EMAIL;
+
+    @SneakyThrows
+    public void sendEmail(MailStructure mailStructure) {
+        verifyReservationDates(mailStructure)
+                .createSimpleEmail(mailStructure);
+
+    }
+
+    private MailSenderService verifyReservationDates(MailStructure mailStructure) throws PastDateException{
+        LocalDate date=LocalDate.now();
+        ConferenceMailStructure conferenceMailStructure= (ConferenceMailStructure) mailStructure;
```

Comment:
The service is intended to be generic, and it should not be based in a specific implementation.
If MailStructure type is anything other than ConferenceMailStructure it will throw and exception.
The service should not be dependend by specific implementation. Please use the interface directly, if you are missing any methods used here, please create an abstract method and implement it on childrens